client/post-editor/editor-action-bar/index.jsx

```diff
@@ -14,28 +15,29 @@ import Tooltip from 'components/tooltip';
 import Button from 'components/button';
 import EditorActionBarViewLabel from './view-label';
 import EditorStatusLabel from 'post-editor/editor-status-label';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPost } from 'state/posts/selectors';
 
-export default React.createClass( {
+class EditorActionBar extends Component {
 
-	displayName: 'EditorActionBar',
-
-	propTypes: {
+	static propTypes = {
 		isNew: React.PropTypes.bool,
 		onPrivatePublish: React.PropTypes.func,
 		post: React.PropTypes.object,
 		savedPost: React.PropTypes.object,
 		site: React.PropTypes.object,
 		type: React.PropTypes.string
-	},
+	};
 
-	getInitialState: function() {
-		return {
-			viewLinkTooltip: false
-		};
-	},
+	state = {
+		viewLinkTooltip: false
+	};
 
 	render() {
 		const multiUserSite = this.props.site && ! this.props.site.single_user_site;
+		const isPasswordProtected = utils.getVisibility( this.props.post ) === 'password';
+		const isPrivate = utils.isPrivate( this.props.post );
```

Comment:
It looks like you're relying on the Redux "Edited Post" to show/hide the `EditorSticky` component. But it looks like, we're still relying on the Flux store for "status" updates (see https://github.com/Automattic/wp-calypso/blob/fix/13446-private-posts-sticky/client/post-editor/editor-visibility/index.jsx#L181). This makes me think we may want to rely on the Flux "Edited Post" instead?